client/state/plugins/installed/actions.js

```diff
@@ -532,3 +535,32 @@ export function fetchSitePlugins( siteId ) {
 export function fetchPlugins( siteIds ) {
 	return ( dispatch ) => siteIds.map( ( siteId ) => dispatch( fetchSitePlugins( siteId ) ) );
 }
+
+export function fetchAllPlugins() {
+	return ( dispatch ) => {
+		dispatch( { type: ALL_PLUGINS_REQUEST } );
+
+		const receivePluginsDispatchSuccess = ( { sites } ) => {
+			dispatch( { type: ALL_PLUGINS_REQUEST_SUCCESS } );
+
+			Object.entries( sites ).map( ( [ siteId, plugins = [] ] ) => {
+				dispatch( receiveSitePlugins( siteId, plugins ) );
+
+				plugins.map( ( plugin ) => {
+					if ( plugin.update && plugin.autoupdate ) {
+						updatePlugin( siteId, plugin )( dispatch );
+					}
+				} );
+			} );
+		};
+
+		const receivePluginsDispatchFail = ( error ) => {
+			dispatch( { type: ALL_PLUGINS_REQUEST_FAILURE, error } );
+		};
+
+		return wpcom.req
+			.get( `/me/sites/plugins` )
```

Comment:
We are using `get_option` (AFAIK, it caches the result, so any further call for that option on the same HTTP request won't hit the DB), and there's no caching layer implemented.

I've installed a random plugin from `http://calypso.localhost:3000/plugins/:site` and waited for the activation request to finish. Then I refreshed the `http://calypso.localhost:3000/plugins/manage/:site` (in a new tab), and my plugin appeared in the list.

Am I testing this correctly?

In theory, there could be a slight delay because we need that sync action (after the plugin is installed) to be triggered, but I'd say this shouldn't be noticed in most cases.

Reply:
Sounds like you're testing it correctly 👍 

However, `get_option()` runs on a shadow (AKA cache) site, and the data there is populated by sync. There's no guarantee that once you perform a plugin action, it will be synced immediately to the cache site. The delay can be substantial when there's a lot of action going on on the site and there's a large queue of actions to be synced. It may be running quickly on your test site, but that's likely because there are no queued actions for syncing, and the plugin actions get synced to the shadow site immediately.

TBH this seems problematic to me, as it can lead to inconsistent behavior and bad UX.

Reply:
Thanks for all your feedback! To avoid going back and forth here, we moved the discussion [here](pbtFFM-23q-p2#comment-2908).

As the sync times are much better now, we can leave this as-is and monitor it closely to see if there will be any inconsistencies.